qemu: uncaught target signal 4 (Illegal instruction) - core dumped

I tried to have fun with instructions with README.md, but got error like the following,

``` bash
$ make
  SHECC	out/shecc-stage2.elf
qemu: uncaught target signal 4 (Illegal instruction) - core dumped
make: *** [Makefile:59: out/shecc-stage2.elf] Illegal instruction (core dumped)
```

Then I made a simple test,
1. Passed to execute native `shecc` binary on host.
2. Failed to execute `shecc-stage2.elf` with `qemu-arm shecc-stage2.elf hello.c` command.

![2020-09-06_101439](https://user-images.githubusercontent.com/3453968/92316650-e4224d00-f029-11ea-99ea-a1b8ba0afa73.png)

--- a/src/arm.c
+++ b/src/arm.c
@@ -256,9 +256,9 @@ int __rsb_i(arm_cond_t cond, arm_reg rd, int imm, arm_reg rn)
     return __mov(cond, 1, arm_rsb, 0, rd, rn, imm);
 }
 
-int __cmp_r(arm_cond_t cond, arm_reg rd, arm_reg r1, arm_reg r2)
+int __cmp_r(arm_cond_t cond, arm_reg r1, arm_reg r2)
 {
-    return __mov(cond, 0, arm_cmp, 1, r1, rd, r2);
+    return __mov(cond, 0, arm_cmp, 1, r1, 0, r2);
 }
 
 int __teq(arm_reg rd)

--- a/src/codegen.c
+++ b/src/codegen.c
@@ -376,7 +376,7 @@ void code_generate()
         case OP_gt:
         case OP_geq:
             /* we want 1/nonzero if equ, 0 otherwise */
-            emit(__cmp_r(__AL, dest_reg, dest_reg, OP_reg));
+            emit(__cmp_r(__AL, dest_reg, OP_reg));
             emit(__zero(dest_reg));
             emit(__mov_i(arm_get_cond(op), dest_reg, 1));
 

